main.py

```diff
@@ -8,34 +8,77 @@
 """
 
 from documentcloud.addon import AddOn
+import os
+from dotenv import load_dotenv
+import heapq
 
+load_dotenv()
+print('NLTK_DATA', os.environ['NLTK_DATA'])
+# .env needs to be loaded before importing nltk so that it knows
+# where to look for data.
+import nltk
 
-class HelloWorld(AddOn):
-    """An example Add-On for DocumentCloud."""
+stopwords = nltk.corpus.stopwords.words('english')
 
-    def main(self):
-        """The main add-on functionality goes here."""
-        # fetch your add-on specific data
-        name = self.data.get("name", "world")
+# https://stackabuse.com/text-summarization-with-nltk-in-python/
+def summarize(doc):
+  sentence_list = nltk.sent_tokenize(doc.full_text)
+
+  word_frequencies = {}
+  for word in nltk.word_tokenize(doc.full_text):
+    if word not in stopwords:
+      if word not in word_frequencies.keys():
+        word_frequencies[word] = 1
+      else:
+        word_frequencies[word] += 1
+
+  maximum_frequency = max(word_frequencies.values())
+
+  for word in word_frequencies.keys():
+    word_frequencies[word] = (word_frequencies[word]/maximum_frequency)
+
+  sentence_scores = {}
+  for sent in sentence_list:
+    for word in nltk.word_tokenize(sent.lower()):
+      if word in word_frequencies.keys():
+        if len(sent.split(' ')) < 30:
+          if sent not in sentence_scores.keys():
+            sentence_scores[sent] = word_frequencies[word]
+          else:
+            sentence_scores[sent] += word_frequencies[word]
 
-        self.set_message("Hello World start!")
+  summary_sentences = heapq.nlargest(7, sentence_scores, key=sentence_scores.get)
+  summary = ' '.join(summary_sentences)
 
-        # add a hello note to the first page of each selected document
-        if self.documents:
-            for document in self.client.documents.list(id__in=self.documents):
-                document.annotations.create(f"Hello {name}!", 0)
-        elif self.query:
-            documents = self.client.documents.search(self.query)[:3]
-            for document in documents:
-                document.annotations.create(f"Hello {name}!", 0)
+  return summary
+
+class Summarize(AddOn):
+    """A document summarization Add-On for DocumentCloud."""
+
+    def add_summary(self, document, file_):
+        summary = summarize(document)
+        file_.write(f"{document.title}\n{document.canonical_url}\n\n{summary}\n\n")
+        self.set_message(f"Summarized {document.canonical_url}.")
+
+    def main(self):
+        """The main add-on functionality goes here."""
+        self.set_message("Starting summarization.")
 
-        with open("hello.txt", "w+") as file_:
-            file_.write("Hello world!")
-            self.upload_file(file_)
+        doc_list = self.client.documents.list(id__in=self.documents)
+        if not doc_list:
+            if self.query:
+                doc_list = self.client.documents.search(self.query)[:3]
```

Comment:
You don't need the `[:3]` here - I had that to prevent people from accidentally adding hello world notes to all of their documents.  If you actually want to be able to run this on all documents in a search, that is fine.

I am also going to add some code to the library to simplify getting documents by either method, so stay tuned for that.

Reply:
Good to know why that was there!